feat: Add category column to blob and transaction tables

xFJA opened the pull request:
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [X] I have filled out the description and linked the related issues.

### Description
A `category` column has been added to `blob`, `block` and `transactions` tables at the init of the rows without any header name

### Related Issue
Closes #595 

### Screenshots 
## Transaction table
<img width="1510" alt="image" src="https://github.com/user-attachments/assets/18de2743-4074-487e-84de-33eb5c10ffc6">

## Block table

https://github.com/user-attachments/assets/a932f886-d5b0-44a1-8c2f-68598a8513ef

PJColombo reviewed (commented):
"Blob Gas Price" column on tx table looks cropped

<img width="181" alt="image" src="https://github.com/user-attachments/assets/7cf285c3-eaac-40e2-b922-4904c62256c7">


Also, category is not being displayed on the blobs table:

<img width="1559" alt="image" src="https://github.com/user-attachments/assets/3e919af0-aa79-44e5-ad50-8a990d290447">

PJColombo commented:
I'm also thinking it would make more sense to remove "Rollup" column and display rollup logo in the "Category" by adding it right next to the "Rollup" label.

xFJA commented:
> "Blob Gas Price" column on tx table looks cropped
> 
> <img alt="image" width="181" src="https://private-user-images.githubusercontent.com/33203511/384436061-7cf285c3-eaac-40e2-b922-4904c62256c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNTU1ODEsIm5iZiI6MTczMTE1NTI4MSwicGF0aCI6Ii8zMzIwMzUxMS8zODQ0MzYwNjEtN2NmMjg1YzMtZWFhYy00MGUyLWI5MjItNDkwNGM2MjI1NmM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDEyMjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzODYxNmI5ZTY0ODcwMTc4YTRkMDc2OGY1YzkwZmZmMjk5NDgzYmUyM2M3ZjE5Nzc5MzI1NTc4ODFjMmQwMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qMKjHw1VPiWHoQK6WmR9dFPJbxoC2-Wm0-n4XBybyzg">
> Also, category is not being displayed on the blobs table:
> 
> <img alt="image" width="1559" src="https://private-user-images.githubusercontent.com/33203511/384436485-3e919af0-aa79-44e5-ad50-8a990d290447.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNTU1ODEsIm5iZiI6MTczMTE1NTI4MSwicGF0aCI6Ii8zMzIwMzUxMS8zODQ0MzY0ODUtM2U5MTlhZjAtYWE3OS00NGU1LWFkNTAtOGE5OTBkMjkwNDQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDEyMjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhM2I2Nzg0MzYzMjY1ZWFlMTdjMjVkYTQ4MzkwZTgzZGIwNzNkZjY2MzJlMWFlYWUxYjgyYjA1MWU3ODU3NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SuQaOKl2_X-X0m_0-1OzFZFwCsxqvRjlWVb6V1cQTuw">



> "Blob Gas Price" column on tx table looks cropped
> 
> <img alt="image" width="181" src="https://private-user-images.githubusercontent.com/33203511/384436061-7cf285c3-eaac-40e2-b922-4904c62256c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNTU1ODEsIm5iZiI6MTczMTE1NTI4MSwicGF0aCI6Ii8zMzIwMzUxMS8zODQ0MzYwNjEtN2NmMjg1YzMtZWFhYy00MGUyLWI5MjItNDkwNGM2MjI1NmM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDEyMjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzODYxNmI5ZTY0ODcwMTc4YTRkMDc2OGY1YzkwZmZmMjk5NDgzYmUyM2M3ZjE5Nzc5MzI1NTc4ODFjMmQwMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qMKjHw1VPiWHoQK6WmR9dFPJbxoC2-Wm0-n4XBybyzg">
> Also, category is not being displayed on the blobs table:
> 
> <img alt="image" width="1559" src="https://private-user-images.githubusercontent.com/33203511/384436485-3e919af0-aa79-44e5-ad50-8a990d290447.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNTU1ODEsIm5iZiI6MTczMTE1NTI4MSwicGF0aCI6Ii8zMzIwMzUxMS8zODQ0MzY0ODUtM2U5MTlhZjAtYWE3OS00NGU1LWFkNTAtOGE5OTBkMjkwNDQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDEyMjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhM2I2Nzg0MzYzMjY1ZWFlMTdjMjVkYTQ4MzkwZTgzZGIwNzNkZjY2MzJlMWFlYWUxYjgyYjA1MWU3ODU3NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SuQaOKl2_X-X0m_0-1OzFZFwCsxqvRjlWVb6V1cQTuw">

- Fixed the category not displayed on `blobs` table 
- I've checked the `Blob Gas Price` column in `blobs` table and I cannot notice any crop. The table has a horizontal scroll for the overflowing columns. Could you check it again?
  <img width="457" alt="image" src="https://github.com/user-attachments/assets/74c38c15-2f95-4b89-920b-d6f26a991484">

xFJA commented:
> I'm also thinking it would make more sense to remove "Rollup" column and display rollup logo in the "Category" by adding it right next to the "Rollup" label.

Agree! Having both columns is redundant and removes width space. I have merged both columns into one with your proposal 👍

PabloCastellano reviewed (approved):
lgtm :+1:

PJColombo commented:
I'll comment here what we discussed on Discord:

Having both the category and icon feels too verbose. As an alternative, I propose the following:

We add a new column as the first column in all tables, leaving the title blank:
	•	If the category is ROLLUP, we display the icon; otherwise, the column remains empty.
	•	For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as needed.

xFJA commented:
> I'll comment here what we discussed on Discord:
> 
> Having both the category and icon feels too verbose. As an alternative, I propose the following:
> 
> We add a new column as the first column in all tables, leaving the title blank: • If the category is ROLLUP, we display the icon; otherwise, the column remains empty. • For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as nee

> I'll comment here what we discussed on Discord:
> 
> Having both the category and icon feels too verbose. As an alternative, I propose the following:
> 
> We add a new column as the first column in all tables, leaving the title blank: • If the category is ROLLUP, we display the icon; otherwise, the column remains empty. • For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as needed.

Agree! Done 👍 

In `blocks` table:
- I added the rollup icons overlapped a little bit
- I added the rollup icon in the inner expand table for each specific transaction

PJColombo commented on apps/web/src/pages/txs.tsx:
```diff
@@ -202,6 +202,14 @@ const Txs: NextPage = function () {
             cells: [
+              {
+                item:
+                  category && category === "rollup" && rollup ? (
```
```suggestion
                 category === "rollup" && rollup ? (
```

PJColombo commented on apps/web/src/pages/blocks.tsx:
```diff
@@ -174,6 +190,24 @@ const Blocks: NextPage = function () {
+                      return tx.rollup ? (
+                        <div
+                          key={i}
+                          className="-ml-1 -translate-y-1 scale-100 first-of-type:ml-0 hover:scale-125 "
```
I'd prefer to remove the scale animation to keep consistency with the rest of tables

xFJA replied:
Done

PJColombo commented on apps/web/src/pages/blocks.tsx:
```diff
@@ -127,12 +133,22 @@ const Blocks: NextPage = function () {
                 cells: [
+                  {
+                    item:
+                      category && category === "rollup" && rollup ? (
```
```suggestion
                      category === "rollup" && rollup ? (
```

PJColombo commented on apps/web/src/pages/blobs.tsx:
```diff
@@ -84,8 +89,19 @@ const Blobs: NextPage = function () {
             cells: [
+              {
+                item:
+                  transaction?.category &&
+                  transaction.category === "rollup" &&
+                  transaction.rollup ? (
```
```suggestion
                  transaction?.category === "rollup" &&
                  transaction.rollup ? (
```

xFJA replied:
Done

PJColombo reviewed (changes requested):
Nice job.

I’ve left a few comments along with some small suggestions.

PJColombo reviewed (approved):
lgtm
